src/feature-flags/hooks.tsx

```diff
@@ -32,3 +35,30 @@ export const IfFeature: React.FC<IfFeatureProps> = ({ flag, children, fallback }
   const isOn = useIsOnFeatureFlag(flag);
   return <>{isOn ? children : fallback ?? null}</>;
 };
+
+export const createConditionsHook = (
+  keys: ConditionKey[],
+): (() => Record<(typeof keys)[number], boolean>) => {
+  return () => {
+    const memoizedKeys = useDeepCompareMemoize(keys);
```

Comment:
No need to use `useDeepCompareMemoize` here because createConditionHook will be called outside of components every time and keys reference will be stable whenever the created hook is called